### ISSUE TITLE
Allow Event Director to manage Age Group

### ISSUE THREAD

**Issue by rdunlop** [rdunlop/unicycling-registration#300]
It should be possible for an Event Director to manage their competitions better.

Currently, Competition Directors are limited in the following ways:
- Cannot Edit the configuration of their competitions
- Cannot create/update/delete Age Groups.

Both of these restrictions should be lifted:
- Should be able to create new age groups
- Should be able to delete any age group which is not assigned to any competition
- Should be able to update any age group which is only assigned to their competitions (and not competitions which are NOT within their control)

- Should be able to Edit their competition, changing any of the variables.
- Should be able to create new competitions (?? this is still un-decided).

**Comment by scotthue** [comment]
"Should be able to delete any age group which is not assigned to any competition" Why? 

"Should be able to Edit their competition, changing any of the variables." I think the edit page needs some more cleanup (or at least hiding non-relevant options) before an Event Director should be given this permission.

Another idea entierly: Currently an Event Director is assigned to a `Category` and `Event` from the Competition Overview page. These things actually come from the registration side and not the competition side. The Event Director should then be able to see the Registration Overview for that event (which currently doesn

**Comment by rdunlop** [comment]
I think that Directors should be able to create new Age Groups...and if they can create an age group, they should be able to delete it.

It sounds like you're saying that the Director role has a bug where it cannot see the Registration Overview for their Events. and also a bug where they cannot update the Best Times  for their events.

I'm hesitating to allow Directors to change Best Times...as I dislike allowing too many people to change Registration data...though I think it's probably the right call.

**Comment by scotthue** [comment]
I agree that creation permissions implies deletion permissions from a logical standpoint, but it is a more dangerous behavior. By not allowing deletion, you are allowing the possibility of superfluous age groups, but not the deletion of something another director was working on.

I believe that is correct. Currently we have had to give "important/capable" directors, like Patricia or Andreas the additional permission of Event Planner in order to let them see event sign ups on the registration overview page (and not just eligible competitions on the competition page).

I see two reasons why Best Times should be changed by someone other than the registrant: either (1) there is a mistake in the format by which it was entered (e.g. minutes and hours mixed up). Or (2) the director wants to change the best time because they believe it is inaccurate or not valid. In the first case, I think the changes should be made on the "registration side" where best times editing is already done, this way the registrant sees their new (correct) best time when they look at their registration. Adding feature #242 would also solve many of these problems. In the second case, this feels more like something that should be done on the "competition side" of things because you are changing the best time from what the registrant _intented_ to enter. Maybe this could be done where all of the best times are imported into the competition details and can be edited, sorted, put into waves, etc here instead of what Connie and I always do in excel. However, I'm not sure how much I like having best times editable in two places. It's confusing.
